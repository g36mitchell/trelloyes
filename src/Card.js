import React from 'react';
import './Card.css';

class CardClass extends React.Component {

    static defaultProps = {
        title: 'New Card',
        content: 'Put your activities here.'
    };

    render() {
 
        return (
            <div className='Card'>
                <button type="button">delete</button>
                <h3>{this.props.title}</h3>
                <p>{this.props.content}</p>
            </div>
        );
    }
}

export default CardClass;