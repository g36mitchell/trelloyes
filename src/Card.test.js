import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card';
import renderer from 'react-test-renderer';

/* Test Suite for Card Component */
describe('Card component', () => {

  /* SMOKE */ 
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Card />, div);
  ReactDOM.unmountComponentAtNode(div);
}); 

it('renders with props', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Card 
        key='a' 
        content='lorem ipsum cat scrabble clawing' 
        title='TITLE 1' />, div);
    ReactDOM.unmountComponentAtNode(div);
});
  
/*  SNAPSHOT */
it('renders the UI as expected with props', () => {
    const tree = renderer
      .create(<Card 
        key='a' 
        content='lorem ipsum cat scrabble clawing' 
        title='TITLE 1' />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
});

});