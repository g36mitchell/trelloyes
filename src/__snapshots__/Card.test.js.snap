// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders the UI as expected with props 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3>
    TITLE 1
  </h3>
  <p>
    lorem ipsum cat scrabble clawing
  </p>
</div>
`;
